c10/macros/Export.h

```diff
@@ -100,15 +100,30 @@
 
 // NB: For now, HIP is overloaded to use the same macro, but ideally
 // HIPify should translate TORCH_CUDA_API to TORCH_HIP_API
-#if defined(TORCH_CUDA_BUILD_MAIN_LIB) || defined(TORCH_HIP_BUILD_MAIN_LIB)
-#define TORCH_CUDA_API C10_EXPORT
-#else
-#define TORCH_CUDA_API C10_IMPORT
+// JX: I removed the || defined(TORCH_HIP_BUILD_MAIN_LIB) check for TORCH_CUDA_*_API
+// since TORCH_HIP_API seems properly initialized below
+// libtorch_cuda_cu.so
+#ifdef TORCH_CUDA_CU_BUILD_MAIN_LIB
+#define TORCH_CUDA_CU_API C10_EXPORT
+#elif defined(BUILD_SPLIT_CUDA)
+#define TORCH_CUDA_CU_API C10_IMPORT
+#endif
+
+// libtorch_cuda_cpp.so
+#ifdef TORCH_CUDA_CPP_BUILD_MAIN_LIB
+#define TORCH_CUDA_CPP_API C10_EXPORT
+#elif defined(BUILD_SPLIT_CUDA)
+#define TORCH_CUDA_CPP_API C10_IMPORT
 #endif
 
-// This is in preparation for the imminent torch_cuda split
-#define TORCH_CUDA_CU_API TORCH_CUDA_API
-#define TORCH_CUDA_CPP_API TORCH_CUDA_API
+// libtorch_cuda.so (where torch_cuda_cu and torch_cuda_cpp are a part of the same api)
+#ifdef TORCH_CUDA_BUILD_MAIN_LIB
+#define TORCH_CUDA_CPP_API C10_EXPORT
+#define TORCH_CUDA_CU_API C10_EXPORT
+#elif !defined(BUILD_SPLIT_CUDA)
```

Comment:
is this to avoid the duplicate definition in buck?

Reply:
Yes!

Reply:
> I see. so per Nikita's comment we don't need to explain this here b/c python setup script will correctly add them. maybe remove the comments about setting env var?

Actually, my comment is to say that you MUST set the env var (instead of only using -DBUILD_SPLIT_CUDA=ON) because the tests will not work if the environment variable isn't set (`cpp_extensions.py` relies on the environment variable being present).

Reply:
Have you tried (could be done in a followup PR) creating a dummy `libtorch_cuda.so` that depens on two parts of split library (which would mimic the pattern of libtorch.so, which doesn't have any functionality on its own but merely links libtorch_cpu and libtorch_cuda together.